### PR TITLE
Generate from the lens

### PR DESCRIPTION
**Pull request by levmckinney**
Adds the ability to easily generate tokens from the tuned lens.

### CODE REVIEW

File: tuned_lens/nn/lenses.py [hunk 1]
```diff
@@ -316,3 +316,66 @@ def forward(self, h: th.Tensor, idx: int) -> th.Tensor:
...
+            do_sample: Whether to use sampling or greedy decoding.
+            temp: The temperature to use for sampling.
+            max_new_tokens: The maximum number of tokens to generate.
+
+        Returns:
+            The prompt concatenated with the newly generated tokens.
+        """
+        eos_token = model.generation_config.eos_token_id
+
+        tokens = input_ids.clone()
```
Comment (rhaps0dy) [inline tuned_lens/nn/lenses.py:346]: There doesn't seem to be any need to clone this ? Maybe then you'd have to add to the smoke test "the input is unmodified".

File: tuned_lens/nn/lenses.py [hunk 2]
```diff
@@ -316,3 +316,66 @@
...
+                use_cache=True,
+                past_key_values=past_key_values,
+            )
+            past_key_values = output.past_key_values
+            hidden = output.hidden_states[layer]
+            new_hidden = hidden[:, -1, :]
+            new_logits = self.forward(new_hidden, layer)
+            if do_sample:
+                new_logits = new_logits / temp
+                new_logits = th.nn.functional.log_softmax(new_logits, dim=-1)
```
Comment (rhaps0dy) [inline tuned_lens/nn/lenses.py:367]: Any reason why you take `log_softmax` and then `.exp()` ? `softmax()` would do the same

File: tuned_lens/nn/lenses.py [hunk 3]
```diff
@@ -316,3 +316,66 @@
...
+                new_tokens = th.multinomial(new_logits.exp(), num_samples=1)
+            else:
+                new_tokens = new_logits.argmax(dim=-1, keepdim=True)
+
+            # Once a sequence has generated an EOS token, it should not generate any
+            # other tokens.
+            done = done | (new_tokens == eos_token)
+            new_tokens = new_tokens.masked_fill(done, eos_token)
```
Comment (rhaps0dy) [inline tuned_lens/nn/lenses.py:375]: Beautiful!

**Review by rhaps0dy — APPROVED** [review]
LGTM, just some nitpicks. Fix CI and then merge.
